### PR TITLE
[BEAM-52] Kafka IO

### PR DESCRIPTION
**Pull request by dhalperi**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [ ] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [x] Replace "<Jira issue #>" in the title with the actual Jira issue
       number, if there is one.
 - [x] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

TODO: @rangadi @dhalperi

- [x] Fix licenses
- [x] Fix names Dataflow -> Beam as in https://github.com/apache/incubator-beam/pull/46.
- [x] Fixup the directory-moving-around stuff as in cd828f1ab007db78fc4dc637bbbf3324897b6d8b and https://github.com/apache/incubator-beam/pull/72.
- [x] Take another quick readthrough.

R: @dpmills @dhalperi

### CODE REVIEW

**Comment by dhalperi** [comment]
Forward-porting https://github.com/GoogleCloudPlatform/DataflowJavaSDK/pull/121. Note that the original author is a Googler (hence covered by CLA) and is still attributed to the actual work.

I fully squashed the original PR and fixed some licenses. More work still in progress.

File: contrib/examples/kafka/pom.xml [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+    limitations under the License.
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+
+  <groupId>com.google.cloud.dataflow</groupId>
+  <artifactId>google-cloud-dataflow-java-contrib-kafka-examples</artifactId>
+  <name>Google Cloud Dataflow Kafka Examples</name>
```
Comment (jbonofre) [inline contrib/examples/kafka/pom.xml]: Maybe we can change to use org.apache.beam Maven coordonates ?

File: contrib/examples/kafka/pom.xml [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+  <description>Examples apps using Kafka Source in Google Cloud Dataflow</description>
+  <version>0.0.1-SNAPSHOT</version>
```
Comment (jbonofre) [inline contrib/examples/kafka/pom.xml]: Version is fine as the KafkaIO can have its own release cycle.

File: contrib/examples/kafka/pom.xml [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+  <packaging>jar</packaging>
+
+  <licenses>
+    <license>
+      <name>Apache License, Version 2.0</name>
```
Comment (jbonofre) [inline contrib/examples/kafka/pom.xml]: License is already define on the parent POM, I don't think it's required to define here again.

File: contrib/examples/kafka/pom.xml [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+  </licenses>
+
+  <properties>
+    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+  </properties>
+
+  <build>
+    <plugins>
+
+      <plugin>
```
Comment (jbonofre) [inline contrib/examples/kafka/pom.xml]: I think it's redundant with the parent pom definition.

File: contrib/examples/kafka/pom.xml (hunk 1 above)
Comment (jbonofre) [inline contrib/examples/kafka/pom.xml]: Anyway, aligned with the other IOs and current discussion, we can move this in the sdks/java/io module later.

**Comment by jbonofre** [comment]
Just couple of small comments (about Maven coordonates and module). Great work anyway !

**Comment by dhalperi** [comment]
Thanks @jbonofre ! We'll take these comments into account as we do the cleanups in the TODO list in the PR description.

File: contrib/examples/kafka/pom.xml (hunk 1 above)
Comment (rangadi) [inline contrib/examples/kafka/pom.xml]: done.

File: contrib/examples/kafka/pom.xml (hunk 2 above)
Comment (rangadi) [inline contrib/examples/kafka/pom.xml]: changed it to '0.1.0-incubating-SNAPSHOT' to match parent.

File: contrib/examples/kafka/pom.xml (hunk 3 above)
Comment (rangadi) [inline contrib/examples/kafka/pom.xml]: done.

File: contrib/examples/kafka/pom.xml (hunk 4 above)
Comment (rangadi) [inline contrib/examples/kafka/pom.xml]: removed.

**Comment by rangadi** [comment]
Looking in Travis failure.

File: .travis.yml [hunk 5]
```diff
@@ -34,5 +34,3 @@ install:
 script:
   - travis_retry mvn -B $MAVEN_OVERRIDE install -U
   - travis_retry travis/test_wordcount.sh
-  - (cd contrib/kafka && mvn -B $MAVEN_OVERRIDE install -U)
```
Comment (rangadi) [inline .travis.yml]: We don't want to test this in travis?

**Comment by rangadi** [comment]
@dhalperi , I made the package renaming changes this morning, only to realize you have already done that..

**Comment by rangadi** [comment]
Thanks Dan! Glad to see KafkaIO is moved out of contrib.

**Comment by dhalperi** [comment]
@rangadi can you give a look and an LGTM? Then I will merge.

Later, we should add an integration test and figure out what to do with the examples. (See comment on https://github.com/apache/incubator-beam/pull/142/commits/1d206630adbebf352606065d19aec7ac97b452ae)

**Comment by rangadi** [comment]
LGTM @dhalperi. Thanks for making the changes.

I agree with your comment on reevaluating if and where example app fits in.

**Pull request closed without merging**